lightning/src/onion_message/messenger.rs

```diff
@@ -933,19 +978,18 @@ where
 	}
 
 	fn handle_onion_message_response<T: OnionMessageContents>(
-		&self, response: Option<T>, reply_path: Option<BlindedPath>, log_suffix: fmt::Arguments
+		&self, response: ResponseInstruction<T>
 	) {
-		if let Some(response) = response {
-			match reply_path {
-				Some(reply_path) => {
-					let _ = self.find_path_and_enqueue_onion_message(
-						response, Destination::BlindedPath(reply_path), None, log_suffix
-					);
-				},
-				None => {
-					log_trace!(self.logger, "Missing reply path {}", log_suffix);
-				},
-			}
+		if let ResponseInstruction::WithoutReplyPath(response) = response {
+			let message_type = response.message.msg_type();
+			let _ = self.find_path_and_enqueue_onion_message(
+				response.message, Destination::BlindedPath(response.reply_path), None,
+				format_args!(
+					"when responding to {} onion message with path_id {:02x?}",
```

Comment:
How about `"when responding with {} to an onion message with path_id {:02x}"`? I think this would read better given the contexts in which the string will be used.

Reply:
I think the second phrasing suits the context better since it emphasizes the `type` of response.

Updated in [b4204b7](https://github.com/lightningdevkit/rust-lightning/commit/b4204b77c2417a5ca337dc2c336ae5d0871d421e)
Thanks a lot for the suggestion!